### PR TITLE
Drop AttributeKey::SCOPE in Rector classes

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/rectorphp/rector/discussions/7924

### CODE REVIEW

File: rules/Strict/Rector/AbstractFalsyScalarRuleFixerRector.php [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
 
 use Rector\Core\Contract\Rector\AllowEmptyConfigurableRectorInterface;
-use Rector\Core\Rector\AbstractRector;
+use Rector\Core\Rector\AbstractScopeAwareRector;
 use Webmozart\Assert\Assert;
 
 /**
  * @see \Rector\Tests\Strict\Rector\BooleanNot\BooleanInBooleanNotRuleFixerRector\BooleanInBooleanNotRuleFixerRectorTest
+ *
+ * @internal
```
Comment (staabm) [inline rules/Strict/Rector/AbstractFalsyScalarRuleFixerRector.php:14]: I would suggest to mark this class internal, so others do not subclass it.

its pretty specific and I think we should protect our users from using it

File: rules/Strict/Rector/ClassMethod/AddConstructorParentCallRector.php [hunk 2]
```diff
@@ -78,19 +79,14 @@ public function getNodeTypes(): array
     /**
      * @param ClassMethod $node
      */
-    public function refactor(Node $node): ?Node
+    public function refactorWithScope(Node $node, Scope $scope): ?Node
     {
-        $classLike = $this->betterNodeFinder->findParentType($node, ClassLike::class);
```
Comment (staabm) [inline rules/Strict/Rector/ClassMethod/AddConstructorParentCallRector.php:83]: re-order conditions: cheap checks first

**Comment by staabm** [comment]
re-triggered CI

**Comment by TomasVotruba** [comment]
Thank you :partying_face:

**Review by samsonasik — APPROVED** [review]

**Pull request merged**